tokio/src/time/clock.rs

```diff
@@ -129,6 +134,49 @@ cfg_test_util! {
         inner.unfrozen = Some(std::time::Instant::now());
     }
 
+    struct AutoAdvanceInhibit(Clock);
+
+    impl Drop for AutoAdvanceInhibit {
+        fn drop(&mut self) {
+            self.0.allow_auto_advance();
+        }
+    }
+
+    /// Temporarily stop auto-advancing the clock (see `tokio::time::pause`)
+    /// and decorate the given future with code to re-enable auto-advance if
+    /// it is dropped before producing a result.
+    ///
+    /// If the future is polled and returns `Ready`, it becomes the caller's
+    /// responsibility to call `allow_auto_advance`. This quirk is to help
+    /// avoid a race between blocking task runners and a paused runtime.
+    pub(crate) fn inhibit_auto_advance<F>(fut: F) -> impl Future<Output = F::Output> + Send + 'static
+    where
+        F: Future + Send + 'static,
+    {
+        // Bump the inhibit count immediately, not inside the async block, to
+        // avoid a race condition when used by spawn_blocking.
+        let guard = clock().map(|clock| {
+            clock.inhibit_auto_advance();
+            AutoAdvanceInhibit(clock)
+        });
+        async move {
+            let result = fut.await;
+            // On success, do not resume auto-advance. It must be done on the
+            // main tokio thread (`BlockingSchedule::release`) to avoid a race.
+            std::mem::forget(guard);
+            result
+        }
```

Comment:
Is this still in use?

Reply:
Yeah, I misunderstood how this was supposed to work. It's gone now.